expression/builtin_compare.go

```diff
@@ -903,7 +913,13 @@ func (c *compareFunctionClass) refineArgs(ctx context.Context, args []Expression
 	arg0IsInt := args[0].GetType().EvalType() == types.ETInt
 	arg1IsInt := args[1].GetType().EvalType() == types.ETInt
 	arg0, arg0IsCon := args[0].(*Constant)
+	if arg0IsCon && arg0.DeferredExpr != nil {
```

Comment:
Deferred constant is still a constant, or compare function will not be refined then used as access condition.